etl/churn.kp/knowledge.md

```diff
@@ -508,19 +520,36 @@ def compute_churn_week(df, week_start, bucket, prefix):
             ),(
                 1,  # active users 
                 x.get('usage_hours', 0),
-                x.get('squared_usage_hours',0) 
+                x.get('squared_usage_hours',0),
+                x.get('total_uri_count', 0),
+                x.get('unique_domains_count', 0)
             )
         )
     )
 
     def reduce_func(x, y):
-        return (x[0] + y[0], # Sum active users
-                x[1] + y[1], # Sum usage_hours
-                x[2] + y[2]) # Sum squared_usage_hours
-
+        return tuple(map(sum, zip(x, y)))
+    
     aggregated = countable.reduceByKey(reduce_func)
 
     records_df = aggregated.map(lambda x: x[0] + x[1]).toDF(record_columns)
+    
+    # Apply some post-processing for other aggregates (i.e. unique_domains_count). This
+    # needs to be done when you want something other than just a simple sum
+    def average(total, n):
+        if not n:
+            return 0.0
+        return float(total)/n
+    average_udf = F.udf(average, DoubleType())
+    
+    # Create new derived columns and any unecessary ones
+    records_df = (
+        records_df
+            .withColumn('average_unique_domains_count',
+                        average_udf(F.col('unique_domains_count'), F.col('n_profiles')))
+            .drop('unique_domains_count')
+            .withColumnRenamed('average_unique_domains_count', 'unique_domains_count')
```

Comment:
I think it would be a good idea to keep the notion of "averaging" in the column name to avoid confusion later. Something like `unique_domains_count_per_profile` would be good (if a little verbose). Some explanatory comments for how the field should be interpreted would help too.